### PR TITLE
fixed ValueError in tensorflow/examples/skflow/mnist_weights.py

### PR DESCRIPTION
**Pull request by tcfuji**
Solution to #3083. 

I replaced `mnist.train.images` with `mnist.train.images.astype(np.float32)` and `mnist.train.labels` with `mnist.train.labels.astype(np.int)`.

### CODE REVIEW

File: tensorflow/examples/skflow/mnist_weights.py [hunk 1]
```diff
@@ -34,8 +35,10 @@
 
 classifier = learn.TensorFlowLinearClassifier(
     n_classes=10, batch_size=100, steps=1000, learning_rate=0.01)
-classifier.fit(mnist.train.images, mnist.train.labels)
-score = metrics.accuracy_score(mnist.test.labels, classifier.predict(mnist.test.images))
+X_train, y_train = mnist.train.images.astype(np.float32), mnist.train.labels.astype(np.int)
```
Comment (rmlarsen) [inline tensorflow/examples/skflow/mnist_weights.py]: Should this be handled in learn.datasets.load_dataset('mnist')?

Comment (martinwicke) [inline tensorflow/examples/skflow/mnist_weights.py]: That would be better. Is the cast necessary?

Comment (rmlarsen) [inline tensorflow/examples/skflow/mnist_weights.py]: From the bug #3083 it appears that the labels should be int32, int64 but are in fact uint8. Perhaps the right fix is for tflearn to seamlessly support uint8 labels?

Comment (martinwicke) [inline tensorflow/examples/skflow/mnist_weights.py]: This is an outdated example, it's in line to be updated, just has to wait its turn. If the new code cannot handle uint8s it should.

**Comment by rmlarsen** [comment]
I'm closing this PR, as we seem to agree that this is not the right fix.
